### ISSUE TITLE
row-names for data tables

### ISSUE THREAD

**Issue by metasoarous** [incanter/incanter#281]
R provides the option of naming rows, which can be quite useful in situations where you want to be able to select matrices/data.frames by non positional index. It seems this could be added and operate somewhat along the lines of col-names. Do you think could be docked for 2.0? I'd be happy to contribute to this as it's something I'd find very useful for stuff I'm working on.

**Comment by metasoarous** [comment]
Definitely keen to work on this. I'll take a look at those namespaces and protocols, and ping you when I'm ready to start hacking. Perhaps we should open up a core.matrix issue for further discussion?

**Comment by mikera** [comment]
Sounds great! Please feel free to open a new core.matrix issue, we can link it to this one so that people can follow

**Comment by metasoarous** [comment]
Looks like there is already an issue for this! https://github.com/mikera/core.matrix/issues/193

**Comment by mikera** [comment]
@metasoarous yes it is definitely related.... I think issue mikera/core.matrix#193 is equivalent to my third bullet above ("Label support added to the main Dataset implementation")
